modules/ting_subsearch/ting_subsearch_suggestions/ting_subsearch_suggestions.module

```diff
@@ -0,0 +1,297 @@
+    '#description' => t('Enter the maximum Levenshtein Distance here, ie. the maximum number of characters to replace, insert or delete to transform the input string into a string from the suggestion data. A higher numbere allows for less similar words to be suggested.'),
+    '#element_validate' => ['element_validate_integer_positive'],
+  ];
+  $form['subsearch']['suggestions']['trigger']['ting_subsearch_suggestions_trigger_limit'] = [
+    '#type' => 'textfield',
+    '#title' => t('Max number of hits to trigger suggestions subsearch'),
+    '#default_value' => variable_get('ting_subsearch_suggestions_trigger_limit', 5),
+    '#description' => t('In order to trigger a subsearch which will provide a suggestion to the user, the user-provided search string must trigger at most this number of hits. NOTE: User-provided searches that trigger 0 hits always trigger a subsearch.'),
+    '#element_validate' => ['element_validate_integer_positive'],
+  ];
+  $form['subsearch']['suggestions']['trigger']['ting_subsearch_suggestions_ps_factor'] = [
+    '#type' => 'textfield',
+    '#title' => t('Ratio between primary and secondary searches for suggestions'),
+    '#default_value' => variable_get('ting_subsearch_suggestions_ps_factor', 10),
+    '#description' => t('Factor between amount of results from primary and secondary search for displaying suggestions.'),
+    '#element_validate' => ['ting_subsearch_factor_field_validate'],
+  ];
+  $form['subsearch']['suggestions']['trigger']['ting_subsearch_suggestions_message_limit'] = [
+    '#type' => 'textfield',
+    '#title' => t('Min number of suggestion search result hits to trigger a suggestion'),
+    '#default_value' => variable_get('ting_subsearch_suggestions_message_limit', 10),
+    '#description' => t("If the suggestion provide lower search result hits than this value don't bother showing a search suggestion."),
+    '#element_validate' => ['element_validate_integer_positive'],
+  ];
+}
+
+/**
+ * Implements hook_ting_search_results_prefix().
+ */
+function ting_subsearch_suggestions_ting_search_results_prefix($keys, $conditions, $results) {
+  $original_search_num_results = $results->getNumTotalObjects();
+
+  // It it's a zero hit search we immediatly look for a suggestion and do
+  // secondary search and redirect if the new search returned any results.
+  if ($original_search_num_results  == 0) {
+    $suggestion = ting_subsearch_suggestions_suggest_keys($keys);
+
+    if ($suggestion) {
+      $suggestion_result = ting_subsearch_do_secondary_search($suggestion);
+
+      if ($suggestion_result->getNumTotalObjects() > 0) {
+        $additional = [
+          'original-search' => $keys,
+        ];
+        $query = ting_subsearch_get_query_string($additional);
+        global $base_url;
+        $url = $base_url . '/search/ting/' . $suggestion . $query;
+
+        header('Location: ' . $url, TRUE, 301);
+        drupal_exit($url);
```

Comment:
No I don't think so. Will use `drupal_goto()` instead

Reply:
I changed it to use `drupal_goto()`. The default response code is 301 and I've left it at that.

Perhaps it could be argued that 302 is more correct, since the configuration can change and the redirect could therefore be considered not permanent?